cip-smart-tx-signing/README.md

```diff
@@ -0,0 +1,82 @@
+
+---
+
+A wallet MUST maintain a cache of signed transactions that have not yet settled on chain. This cache MAY have a maximum size, and if the cache is full, the wallet MAY remove transactions from the cache, beginning with the oldest transaction to store a newer transaction. The wallet MAY remove cached transactions once they are seen on chain.
+
+If a user signs a transaction through the [CIP-30 signTx](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0030/README.md#apisigntxtx-cbortransaction-partialsign-bool--false-promisecbortransaction_witness_set) endpoint, the [CIP-103? signTxs](https://github.com/cardano-foundation/CIPs/blob/508ea0557bcd17d793da90312789165dcef8a4db/CIP-0103/README.md#apicip103signtxstxs-transactionsignaturerequest-promisecbortransaction_witness_set) endpoint, or any other CIP-30 extension that allows signing transactions, the wallet MUST cache the signed transaction with a TTL (time-to-live) greater than or equal to the [TTL of the transaction](https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/babbage/impl/cddl-files/babbage.cddl#L57C8-L57C9). The wallet MUST NOT remove the transaction from the cache until the TTL has expired, the transaction has been seen on chain, or the cache is full and the wallet needs to remove transactions to make room for newer transactions.
+
+When displaying a transaction summary to the user, the wallet MUST search for the outputs referenced by inputs in the following order:
+
+1. The latest known on-chain UTxO set
+2. The cache of signed transactions
+3. The set of transactions being signed
+
+If a transaction contains an input that references an output that cannot be found, the wallet MUST display a warning to the user that the transaction includes unidentifiable inputs. The style and wording of this warning is left up to the wallet. The wallet MAY prevent the user from signing a transaction with unidentified inputs, but this may cause issues when interacting with some dApps.
+
+## Rationale: how does this CIP achieve its goals?
+
+The proposed approach allows wallets to provide a complete and accurate summary of transactions without prohibiting transaction chaining or requiring dApps to change the way they interface with wallets.
+
+Although it cannot entirely reduce the risk of a malicious actor tricking a user into signing a transaction that they didn't intend to, it will hopefully reduce the risk by showing an accurate summary of the transaction more often and, when not possible, displaying a clear warning to the user.
+
+## Path to Active
+
+### Acceptance Criteria
+
+In order for this standard to be considered 'Active', the following criteria must be met:
+
+- [ ] A thorough review by all relevant stakeholders
+- [ ] Implementation in at least two Cardano wallets
+
+### Implementation Plan
+
+N/A
```

Comment:
What about demonstrating this in the extension namespace as a reference implementation?

Reply:
The plan was to make this a fundamental change to the web-wallet bridge standards themselves. Leaving CIP-30's `signTx` and CIP-103's `signTxs` endpoint as is would leave the attack vector available malicious users and render this CIP largely unhelpful.

I am happy to rewrite the CIP to clarify that. I tried to make it clear that, if a wallet implements this CIP, the wallet should implement transaction caching for _any_ current or future transaction signing functionality.

Reply:
I wouldn't suggest rewriting the CIP... this question has just been for my own understanding 🙏